src/basicfuns.jl

```diff
@@ -308,10 +308,25 @@ $(SIGNATURES)
 Return `log(exp(x) + exp(y))`, avoiding intermediate overflow/undeflow, and handling
 non-finite values.
 """
-function logaddexp(x::Real, y::Real)
-    # ensure Δ = 0 if x = y = ± Inf
-    Δ = x == y ? zero(x - y) : abs(x - y)
-    max(x, y) + log1pexp(-Δ)
+logaddexp(x::Real, y::Real) = logaddexp(float.(promote(x, y))...)
```

Comment:
We can avoid the heavy broadcasting machinery here:

```suggestion
logaddexp(x::Real, y::Real) = logaddexp(map(float, promote(x, y))...)
```